### PR TITLE
Fix for GDScript language

### PR DESCRIPTION
**Pull request by williamd1k0**
This is a "partial" fix for #1528 (*.gd files mis-recognized as GAP).

### Added a heuristic rule to match 2 common expressions in GDScript

* `extends` expression:
```gdscript
# Inherit/extend a globally available class
extends SomeClass

# Inherit/extend a named class file
extends "somefile.gd"

# Inherit/extend an inner class in another file
extends "somefile.gd".SomeInnerClass
```

* `class` expression:
```gdscript
class SomeClass:
```

I've tested these rules with some GDScript and GAP repos.

### Added a color to GDScript

* Language color from [official logo](https://godotengine.org/themes/godotengine/assets/logo.svg).

### CODE REVIEW

**Comment by williamd1k0** [comment]
Travis-CI failing in TestColorProximity 😕 
```bash
The following 3 languages have failing color thresholds. Please modify the hex color.
- GDScript (#478cbf) is too close to ["6594b9", "3F85AF", "478cbf", "0096D8", "198CE7"]
- Processing (#0096D8) is too close to ["478cbf", "0096D8"]
- R (#198CE7) is too close to ["478cbf", "198CE7"]
```

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -1339,6 +1339,7 @@ GDB:
   language_id: 122
 GDScript:
   type: programming
+  color: "#478cbf"
```
Comment (brandonblack) [inline lib/linguist/languages.yml]: The hex color here isn't visually distinct enough from other languages already supported by linguist. If these are too close, they make the repository breakdown graph at the top of your repo on github confusing.

You can either revert this commit, or you'll need to update this color to something that doesn't conflict with existing supported languages.

Comment (williamd1k0) [inline lib/linguist/languages.yml]: Yep, I see...
Unfortunately this is the official color 😞 
I think I'll revert this to merge the heuristic rule only.

**Comment by jeremi360** [comment]
[Here another example](https://github.com/jeremi360/Ren-GD/search?l=gap ) of mis classifications my GDScripts as GAP and event _.gittributes_ with`*.gd linguist-language=GDScript` do fix this :(

**Comment by lildude** [comment]
Nudge @williamd1k0. Are you able/willing to address @pchaigno's https://github.com/github/linguist/pull/3478#pullrequestreview-23146668 ?

**Comment by lildude** [comment]
This PR has been pending changes for nearly a year now. It's already marked as stale and will be closed in a week's time if no further updates.

**Comment by lildude** [comment]
No further updates received so I'm closing this PR. Feel free to re-open when you'v got the chance to whip it into shape.

**Pull request closed without merging**